### PR TITLE
Feature/jameszhang878/manage page backend integration

### PR DESCRIPTION
**Pull request by JamesZhang878**
### Administrative Info
[GitHub Issue #48 ]

### Changes
What changes did you make?
- Integrate routes to get users from backend
- Adjust user table and user list to fit backend user schema
- Allow for admins to verify regular users through the manage page

### Testing
How did you confirm your changes worked? 
- I tested my change by manually creating accounts on the backend, and I would then try to verify users with an admin account.

### Confirmation of Change 

https://user-images.githubusercontent.com/73041370/156865438-3af8972b-a84c-438b-a0f5-cda42f2c41af.mov

### CODE REVIEW

File: backend/routes/UserRoutes.js [hunk 1]
```diff
@@ -32,16 +32,11 @@ function checkCurrentUserIsAdmin(req, res, next) {
...
-      next(e);
-    }
-  } else {
-    // prevent route from hanging if no query param passed in
-    res.status(400).send("No query param passed in to users route");
-  }
+  checkCurrentUserIsAdmin(req, res, () => {
+    UserModel.find()
+      .then((users) => res.status(200).json({ users }))
+      .catch((e) => res.status(400).send("An error occurred fetching users"));
```
Comment (Cubified) [inline backend/routes/UserRoutes.js:38]:
```suggestion
      .catch((e) => res.status(400).json({ error: "An error occurred fetching users" }));
```

File: frontend/src/auth.js [hunk 2]
```diff
@@ -158,6 +158,18 @@ async function api_category_download(id) {
   return res;
 }
 
+async function api_get_users() {
+  const res = await api_call("/user/users", {
```
Comment (Cubified) [inline frontend/src/auth.js:162]: To keep this consistent, we should move the URL into `constants/links.js` and reference it here.

File: frontend/src/auth.js [hunk 3]
```diff
@@ -158,6 +158,18 @@ async function api_category_download(id) {
...
+    method: "GET",
+  });
+  return res;
+}
+
+async function api_verify_user(id) {
+  const res = await api_call(`/user/verify/${id}`, {
```
Comment (Cubified) [inline frontend/src/auth.js:169]: (Same deal here)

File: frontend/src/auth.js [hunk 4]
```diff
@@ -158,6 +158,18 @@ async function api_category_download(id) {
...
+    method: "PUT",
+  });
```
Comment (Cubified) [inline frontend/src/auth.js:171]:
```suggestion
  });
  return res;
```

File: frontend/src/components/UserManage.js [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+
+  const updateMyData = (rowIndex, columnId, value) => {
+    // We also turn on the flag to not reset the page
+    setUserData((old) =>
+      old.map((row, index) => {
+        if (index === rowIndex) {
+          // verify user here, reload if fail
+          api_verify_user(row._id).catch((err) => {
+            window.reload();
+          });
```
Comment (Cubified) [inline frontend/src/components/UserManage.js:132]:
```suggestion
          const res = await api_verify_user(row._id);
          if(!res || res.error) window.reload();
```

File: frontend/src/components/UserCardList.js [hunk 6]
```diff
@@ -1,27 +1,31 @@
...
     <div className="user_card" key={Math.random()}>
       <div className="card_col">
-        <button
+        <a
           className="card_item_top"
           aria-label="user_profile"
-          type="button"
-          onClick={() => handleNameClick()}
+          target="blank_"
+          href={`/profile/${user._id}`}
```
Comment (Cubified) [inline frontend/src/components/UserCardList.js:12]:
```suggestion
          href={`${SITE_PAGES.PROFILE}/${user._id}`}
```

File: frontend/src/components/UserCardList.js [hunk 7]
```diff
@@ -1,27 +1,31 @@
 import React, { useState } from "react";
```
Comment (Cubified) [inline frontend/src/components/UserCardList.js:1]:
```suggestion
import React, { useState } from "react";
import { SITE_PAGES } from "../constants/links";
```

File: frontend/src/components/UserManage.js [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+
+const headers = [
+  {
+    Header: "Name",
+    accessor: "name",
+    Cell: ({ row, value }) => (
+      <a
+        className="user_link"
+        target="_blank"
+        href={`/profile/${row.original._id}`}
```
Comment (Cubified) [inline frontend/src/components/UserManage.js:83]:
```suggestion
        href={`${SITE_PAGES.PROFILE}/${row.original._id}`}
```

File: frontend/src/components/UserManage.js [hunk 9]
```diff
@@ -0,0 +1,185 @@
+import React, { useState, useEffect } from "react";
+import Modal from "react-modal";
+import ScrollContainer from "react-indiana-drag-scroll";
+import UserList from "./UserList";
+import UserCardList from "./UserCardList";
+import AssignBtn from "./AssignBtn";
+import { api_get_users, api_verify_user } from "../auth";
```
Comment (Cubified) [inline frontend/src/components/UserManage.js:7]:
```suggestion
import { SITE_PAGES } from "../constants/links";
import { api_get_users, api_verify_user } from "../auth";
```

**Review by Cubified — COMMENTED** [review]
Looks good!  Made a few comments, most are simple one-line changes but one or two are larger (e.g. the global filter being reset after allowing a user access).

**Review by PatrickBrown1 — APPROVED** [review]

File: frontend/src/components/UserCardList.js [hunk 10]
```diff
@@ -1,27 +1,32 @@
...
   return (
     <div className="user_card" key={Math.random()}>
       <div className="card_col">
-        <button
+        <Link
           className="card_item_top"
           aria-label="user_profile"
-          type="button"
-          onClick={() => handleNameClick()}
+          to={`${SITE_PAGES.PROFILE}/${user._id}`}
```
Comment (Cubified) [inline frontend/src/components/UserCardList.js:13]:
```suggestion
          to={`${SITE_PAGES.PROFILE}/${user._id}`}
          target="blank"
```

File: frontend/src/styles/UserCardList.css [hunk 11]
```diff
@@ -16,19 +23,14 @@
...
 .toggle_btn {
   background-color: transparent;
   border: 0px;
-  font-family: Lato;
-  font-style: normal;
-  font-weight: normal;
+
   font-size: 15px;
   line-height: 18px;
   text-align: center;
```
Comment (Cubified) [inline frontend/src/styles/UserCardList.css:36]:
```suggestion
  text-align: center;
  font-family: "Lato", sans-serif;
```

File: frontend/src/styles/UserCardList.css [hunk 12]
```diff
@@ -78,10 +75,13 @@
 .card_col {
   display: flex;
   flex-direction: column;
-  justify-content: center;
+  justify-content: space-between;
   padding-left: 10px;
+  height: 65px;
```
Comment (Cubified) [inline frontend/src/styles/UserCardList.css:80]:
```suggestion
  height: 65px;
  width: 100%;
```

File: frontend/src/styles/UserCardList.css [hunk 13]
```diff
@@ -78,10 +75,13 @@
 .card_col {
   display: flex;
   flex-direction: column;
-  justify-content: center;
+  justify-content: space-between;
   padding-left: 10px;
```
Comment (Cubified) [inline frontend/src/styles/UserCardList.css:79]:
```suggestion
  padding-left: 30px;
```

File: frontend/src/styles/UserCardList.css [hunk 14]
```diff
@@ -39,9 +41,6 @@
 .toggle_btn_selected {
   background-color: transparent;
   border: 0px;
-  font-family: Lato;
-  font-style: normal;
-  font-weight: normal;
   font-size: 15px;
   line-height: 18px;
   text-align: center;
```
Comment (Cubified) [inline frontend/src/styles/UserCardList.css:46]:
```suggestion
  text-align: center;
  font-family: "Lato", sans-serif;
```

**Review by Cubified — APPROVED** [review]
Sorry to keep this PR in limbo for so long, the only suggestions I made are super minor UX changes.  The rest LGTM!

**Comment by PatrickBrown1** [comment]
Trying to merge but the lints aren't running

**Pull request merged**